spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/simple/InstanceProperties.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2012-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.client.discovery.simple;
+
+import java.net.URI;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import org.jspecify.annotations.Nullable;
+
+import org.springframework.cloud.client.DefaultServiceInstance;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.core.style.ToStringCreator;
+import org.springframework.util.Assert;
+
+public class InstanceProperties {
+
+	private @Nullable String instanceId;
+
+	private @Nullable String serviceId;
+
+	private @Nullable String host;
+
+	private int port;
```

Comment:
How is `port` and `secure` not `@Nullable`?  They have no default value and are not set in the default constructor

Reply:
🤦‍♂️